File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if(nums.size()<2)return nums.size();
        int max1=0;
        int count=1;
        sort(nums.begin(),nums.end());
        for(int i=1;i<nums.size();i++)
        {
           if(nums[i]==nums[i-1])continue;
           else if(nums[i]==nums[i-1]+1)
            {
                count++;
            }
            else 
            {
                max1=max(max1,count);
                count=1;
            }
        }
        return  max(max1,count);
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
       //if(nums.length < 2) return nums.length;
       Set<Integer> s = new HashSet<>();
       for(int i : nums){
            s.add(i);
       }
       if(s.size() <= 1) return s.size();
       int max = 0; 
       for(int i : s){
            int length = 1;
            if(!(s.contains(i - 1))){
               length = 1;
            }
            while(s.contains(i + length)){
                length++;
               
            }
            max = Math.max(max, length);
       }
       return max;
    }
}